node/directives/withCurrentProfile.ts

```diff
@@ -185,3 +152,67 @@ function isValidCallcenterOperator(context: Context, email: string) {
 function isLogged(currentProfile: CurrentProfile | null) {
   return currentProfile && currentProfile.email
 }
+
+async function checkUserAccount(
+  context: Context,
+  userProfile: CurrentProfile,
+  resolverInfo: GraphQLResolveInfo
+) {
+  const {
+    dataSources: { identity },
+    vtex: { adminUserAuthToken, storeUserAuthToken, account },
+  } = context
+
+  if (!adminUserAuthToken && !storeUserAuthToken) {
+    throw new AuthenticationError('')
+  }
+
+  const tokenUser = await identity.getUserWithToken(
+    storeUserAuthToken! ?? adminUserAuthToken!
+  )
+
+  if (tokenUser && 'id' in tokenUser && tokenUser.account !== account) {
+    const {
+      fieldName,
+      returnType,
+      operation: { operation, name },
+    } = resolverInfo
+
+    const {
+      vtex: { logger, host },
+      req: {
+        headers: { 'user-agent': userAgent, referer },
+      },
+    } = context
+
+    const logData = {
+      host,
+      referer,
+      userAgent,
+      message: 'Type: CrossTokenAccount',
+      account,
+      tokenAccount: tokenUser.account ?? '',
+      caller: tokenUser.user ?? '',
+      fieldName,
+      fieldType: (returnType as any).name,
+      operation: name?.value ? `${operation} ${name?.value}` : operation,
+    }
+
+    logger.warn(logData)
+  }
+
+  const allowedAccount = await allowedAccounts(
+    context,
+    'account' in tokenUser ? tokenUser.account : ''
```

Comment:
If it is `''` should we still run the `allowedAccounts` function?

Reply:
Maybe initializing `allowedAccount` as `false`, then running this function only if `'account' in tokenUser`? Or something like this.

Reply:
I changed it to a ternary, if true then check isTrustedAccount else false